numba/parfors/parfor_lowering.py

```diff
@@ -888,7 +888,10 @@ def _hoist_internal(inst, dep_on_param, call_table, hoisted, not_hoisted,
 
     uses = set()
     visit_vars_inner(inst.value, find_vars, uses)
+    unhoistable = {assgn.target.name for assgn, _reason in not_hoisted}
+    use_unhoist = uses & unhoistable
     diff = uses.difference(dep_on_param)
+    diff |= use_unhoist
```

Comment:
@DrTodd13, I need your expertise here. I only did the smallest change I can think of. 

The origin is that a `static_getitem(array, index)` is referring to an `array` that is previously placed into `not_hoisted`. The source code looks like:

```python
for .. in loop:
    array = solve(arg)
    res = array[1]     # the static getitem
    if compute(res):
          break
use(array)
```

Pre-patch logic will cause that `static_getitem` to be hoisted out of the loop. I changed it to look into the list of `not_hoisted` and ban hoisting on `uses` that refer to it.